monitoring/notification/dashboard.py

```diff
@@ -0,0 +1,521 @@
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    targets=[Target(
+        expr='\n'.join([
+            'sum by(eventType) (',
+            'rate(',
+            '    notification_queue_processor_processed_events[$__rate_interval]',
+            '))',
+        ]),
+        legendFormat='{{eventType}}',
+    )])
+
+processed_events_in_time = TimeSeries(
+    title='Total Processed Oplog Events',
+    description='Number of processed oplog events by the notification producer',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='hidden',
+    targets=[Target(
+        expr='\n'.join([
+            'sum(',
+            'rate(',
+            '    notification_queue_populator_processed_events[$__rate_interval]',
+            '))',
+        ]),
+        legendFormat='{{target}}',
+    )])
+
+bandwith_per_target = TimeSeries(
+    title='Total Bandwidth',
+    description='Total Bandwidth used to send notifications to targets',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.BYTES,
+    targets=[Target(
+        expr='\n'.join([
+            'sum by(target) (',
+            '    notification_queue_processor_notification_size',
+            ')',
+        ]),
+        legendFormat='{{target}}',
+    )])
+
+bandwith_rate_per_target = TimeSeries(
+    title='Bandwidth',
+    description='Bandwidth used to send notifications to targets in a given time',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.BYTES,
+    targets=[Target(
+        expr='\n'.join([
+            'sum by(target) (',
+            'rate(',
+            '    notification_queue_processor_notification_size[$__rate_interval]',
+            '))',
+        ]),
+        legendFormat='{{target}}',
+    )])
+
+#----------------------------------------------------------------
+
+producer_processing_time = TimeSeries(
+    title='Notification Producer Processing Time',
+    description='Time it takes for the notification producer to process a batch of oplog events',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    lineWidth=3,
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='\n'.join([
+            'avg(',
+            'rate(',
+            '    queue_populator_processing_delay_sec{origin="notification"}[$__rate_interval]',
```

Comment:
I don't think this formula gives the expected value.

Durations metrics should use histogram/summary type, so that:
* we get easily the cumulated duration and number of events (allowing to compute the average duration: total/count)
* we get metrics by buckets, allowing to see how the distribution of processing time and compute percentiles